From 7b2871ef65f9b1ec724f452713c479ea5ff67b60 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:17 -0400
Subject: [fs] ext4: use ext4_update_i_disksize instead of opencoded ones

Message-id: <1414600485-1929-28-git-send-email-lczerner@redhat.com>
Patchwork-id: 98559
O-Subject: [RHEL 7.1 PATCH v2 27/55] ext4: use ext4_update_i_disksize instead of opencoded ones
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Dmitry Monakhov <dmonakhov@openvz.org>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit ee124d2746250786b306952bb8955d3171fa8e69

 Signed-off-by: Dmitry Monakhov <dmonakhov@openvz.org>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/inode.c b/fs/ext4/inode.c
index b0a95a2..344e909 100644
--- a/fs/ext4/inode.c
+++ b/fs/ext4/inode.c
@@ -2654,10 +2654,7 @@ static int ext4_da_write_end(struct file *file,
  if (copied && new_i_size > EXT4_I(inode)->i_disksize) {
   if (ext4_has_inline_data(inode) ||
       ext4_da_should_update_i_disksize(page, end)) {
-   down_write(&EXT4_I(inode)->i_data_sem);
-   if (new_i_size > EXT4_I(inode)->i_disksize)
-    EXT4_I(inode)->i_disksize = new_i_size;
-   up_write(&EXT4_I(inode)->i_data_sem);
+   ext4_update_i_disksize(inode, new_i_size);
    /* We need to mark inode dirty even if
     * new_i_size is less that inode->i_size
     * bu greater than i_disksize.(hint delalloc)
-- 
1.7.1